src/pytest_memray/marks.py

```diff
@@ -46,15 +51,84 @@ def long_repr(self) -> str:
         return f"Test was limited to {max_memory_str} but allocated {total_memory_str}"
 
 
-def limit_memory(
-    limit: str, *, _allocations: list[AllocationRecord]
-) -> Optional[_MemoryInfo]:
+@dataclass
+class _LeakedInfo:
+    """Type that holds all memray-related info for a failed test."""
+
+    max_memory: float
+    total_allocated_memory: int
+    allocations: list[AllocationRecord]
+
+    @property
+    def section(self) -> PytestSection:
+        """Return a tuple in the format expected by section reporters."""
+        total_memory_str = sizeof_fmt(self.total_allocated_memory)
+        max_memory_str = sizeof_fmt(self.max_memory)
+        text_lines = [
+            f"Test leaked {total_memory_str} out of limit of {max_memory_str}",
+            "List of leaked allocations: ",
+        ]
+        for record in self.allocations:
+            size = record.size
+            stack_trace = record.stack_trace()
+            if not stack_trace:
+                continue
+            (function, file, line), *_ = stack_trace
+            if "pytest_memray" in file:
+                # Do not report leaks that happen originated from the plugin
+                continue
+            text_lines.append(f"\t- {function}:{file}:{line} -> {sizeof_fmt(size)}")
+        return "memray-leaked-memory", "\n".join(text_lines)
+
+    @property
+    def long_repr(self) -> str:
+        """Generate a longrepr user-facing error message."""
+        total_memory_str = sizeof_fmt(self.total_allocated_memory)
+        max_memory_str = sizeof_fmt(self.max_memory)
+        return (
+            f"Test was allowed to leak {max_memory_str} but leaked {total_memory_str}"
+        )
+
+
+def limit_memory(limit: str, *, results_file: Path) -> Optional[_MemoryInfo]:
     """Limit memory used by the test."""
+    reader = FileReader(results_file)
+    allocations = list(
+        (reader.get_high_watermark_allocation_records(merge_threads=True))
+    )
     max_memory = parse_memory_string(limit)
-    total_allocated_memory = sum(record.size for record in _allocations)
+    total_allocated_memory = sum(record.size for record in allocations)
     if total_allocated_memory < max_memory:
         return None
-    return _MemoryInfo(max_memory, total_allocated_memory, _allocations)
+    return _MemoryInfo(max_memory, total_allocated_memory, allocations)
+
+
+def check_leaks_runner(
+    func: Callable[[], Any],
+    results_file: Path,
+    *marker_args: Any,
+    warmups: int = 1,
+    **marker_kwargs: Any,
+) -> Any:
+    result = None
+    for _ in range(warmups):
+        with Tracker("/dev/null"):
```

Comment:
Is the only reason we use a `Tracker` here to force 3.11 to allocate its line number table for the trace function? If so, we could get the same effect by installing a no-op profile function, which would cost less overhead.